web/src/settings_components.ts

```diff
@@ -1649,6 +1670,56 @@ export function create_realm_group_setting_widget({
     });
 }
 
+type stream_setting_name = "can_remove_subscribers_group";
+
+export function create_stream_group_setting_widget({
+    $pill_container,
+    setting_name,
+    sub,
+}: {
+    $pill_container: JQuery;
+    setting_name: stream_setting_name;
+    sub?: StreamSubscription;
+}): GroupSettingPillContainer {
+    const pill_widget = group_setting_pill.create_pills($pill_container, setting_name, "stream");
+    const opts: {
+        setting_name: string;
+        sub: StreamSubscription | undefined;
+        setting_type: "stream";
+    } = {
+        setting_name,
+        sub,
+        setting_type: "stream",
+    };
+    group_setting_pill.set_up_pill_typeahead({pill_widget, $pill_container, opts});
+
+    if (sub !== undefined) {
+        group_setting_widget_map.set(setting_name, pill_widget);
```

Comment:
Do we have to worry about collisions in `group_setting_widget_map`?

Reply:
I guess you are talking about collisions for different streams since setting names are unique as of now. We show edit UI for only one stream at a time so that should be fine, like we have for groups.

Reply:
Yeah, I guess they're unique now, but you could imagine us adding settings with matching names. Anyway we might want to think about making `group_setting_widget_map` have a 2-tier structure, with first being the type, or something.